polkadot/node/subsystem-types/src/messages.rs

```diff
@@ -473,6 +473,8 @@ pub enum AvailabilityRecoveryMessage {
 		CandidateReceipt,
 		SessionIndex,
 		Option<GroupIndex>, // Optional backing group to request from first.
+		Option<CoreIndex>,  /* Optional core index that the candidate was occupying. Needed for
```

Comment:
> we do regular validator index to chunk index mapping

actually, we don't do any mapping. The mapping is done by the validators that respond to our peers, since we request by validator index.

Indeed, if it's None, we cannot attempt systematic recovery. I'll explain a bit more in the comment